devito/dimension.py

```diff
@@ -31,10 +30,38 @@ def __str__(self):
     @property
     def symbolic_size(self):
         """The symbolic size of this dimension."""
-        try:
-            return Number(self.size)
-        except TypeError:
-            return self.rtargs[0].as_symbol
+        return self.rtargs[0].as_symbol
+
+    @property
+    def size(self):
+        return None
+
+
+class FixedDimension(FixedDimensionArgProvider, Dimension):
+
+    is_Fixed = True
+    """This class defines the behaviour of a dimension whose size is fixed
+       at the time of problem definition and can thus be baked into generated
+       code
+    """
+
+    def __new__(cls, name, **kwargs):
+        newobj = super(FixedDimension, cls).__new__(cls, name)
+        newobj._size = kwargs.get('size', None)
+        return newobj
+
+    @property
+    def symbolic_size(self):
+        """The symbolic size of this dimension."""
+        return Number(self.size)
```

Comment:
This is the only difference with a non-fixed dimension (unless I am missing something). Wouldn't be easier/nicer to only have a "conditional" property?